src/Build/BackEnd/Shared/TargetResult.cs

```diff
@@ -284,6 +280,68 @@ private void RetrieveItemsFromCache()
             }
         }
 
+        private void TranslateItems(ITranslator translator)
+        {
+            var itemsCount = _items?.Length ?? 0;
+            translator.Translate(ref itemsCount);
+
+            if (translator.Mode == TranslationDirection.WriteToStream)
+            {
+                // We will just calculate a very rough starting buffer size for the memory stream based on the number of items and a
+                // rough guess for an average number of bytes needed to store them compressed.  This doesn't have to be accurate, just
+                // big enough to avoid unnecessary buffer reallocations in most cases.
+                var defaultCompressedBufferCapacity = _items.Length * 64;
+                // Again, a rough calculation of buffer size, this time for an uncompressed buffer.  We assume compression 
+                // will give us 2:1, as it's all text.
+                var defaultUncompressedBufferCapacity = defaultCompressedBufferCapacity * 2;
+                using var itemsStream = new MemoryStream(defaultUncompressedBufferCapacity);
+                var itemTranslator = BinaryTranslator.GetWriteTranslator(itemsStream);
+
+                // When creating the interner, we use the number of items as the initial size of the collections since the
+                // number of strings will be of the order of the number of items in the collection.  This assumes basically
+                // one unique string per item (frequently a path related to the item) with most of the rest of the metadata
+                // being the same (and thus interning.)  This is a hueristic meant to get us in the ballpark to avoid 
+                // too many reallocations when growing the collections.
+                var interner = new LookasideStringInterner(StringComparer.Ordinal, _items.Length);
+                foreach (TaskItem t in _items)
+                {
+                    t.TranslateWithInterning(itemTranslator, interner);
+                }
+
+                interner.Translate(translator);
+                if (itemsStream.TryGetBuffer(out var buffer))
+                {
+                    var offset = buffer.Offset;
```

Comment:
Is this really needed? We know how we created the `MemoryStream` so `Offset` is guaranteed to be 0.

Reply:
Removed